main.rb

```diff
@@ -0,0 +1,132 @@
+# rubocop:disable Metrics/CyclomaticComplexity,Metrics/PerceivedComplexity
+
+module Enumerable
+  def my_each
+    return to_enum(__method__) unless block_given?
+
+    my_iterable = enforce_arr
+    arr_len = my_iterable.length
+    arr_len.times do |index|
+      yield(my_iterable[index])
+    end
+    self
+  end
```

Comment:
- [ ] When `my_each` is called on a hash and a block is given, it should call the block once for each element in the hash
    - it appears as though yours does not return a hash.